File: solutions/python/perfect-numbers/1/perfect_numbers.py
def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if number <=0:
        raise ValueError("Classification is only possible for positive integers.")
    else:
            sum_of_divisors = 0

    for i in range(1, number):
        if number % i == 0:
            sum_of_divisors += i

    if sum_of_divisors == number:
        return "perfect"

    elif sum_of_divisors > number:
        return "abundant"

    else:
        return "deficient"

File: solutions/javascript/perfect-numbers/1/perfect-numbers.js
//
// This is only a SKELETON file for the 'Perfect Numbers' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const classify = (n) => {
  if (n<=0){
  throw new Error('Classification is only possible for natural numbers.');
  }
  let sum =0;
    for (let i=1;i<=n-1;i++){
      if(n%i===0){
        sum+=i;
      }}
  if (sum === n) return "perfect";
if (sum > n) return "abundant";
return "deficient";
      

};
